handled the priority of ACL ruleprops to be case insensitive

madhanmellanox opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines:
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "closes #xxxx",
"fixes #xxxx" or "resolves #xxxx" so that GitHub automatically closes the related
issue when the PR is merged.

If you are adding/modifying/removing any command or utility script, please also
make sure to add/modify/remove any unit tests from the sonic-utilities-tests
directory as appropriate.

If you are modifying or removing an existing 'show', 'config' or 'sonic-clear'
subcommand, or you are adding a new subcommand, please make sure you also
update the Command Line Reference Guide (doc/Command-Reference.md) to reflect
your changes.

Please provide the following information:
-->

**- What I did**
I made the priority of ACL rule props in the ACL commands to be case insensitive.

**- How I did it**
Instead of checking the ACL ruleprops dictionary for "PRIORITY" based key, I made it the actual key to be searched for.

**- How to verify it**
I verified the fix by testing on my switch using different case sensitive options of keyword "Priority".

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

Commit:
handled the priority of ACL ruleprops to be case insensitive

stephenxs committed:
[config] Add sflow to _reset_failed_services (#850)

nazariig committed:
[fwutil]: Use overlay driver when mounting next image filesystem (#825)

* [fwutil]: Use overlay driver when mounting next image filesystem.

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

* [fwutil]: Update command reference.

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

shlomibitton committed:
Fix for adding L3 interface to Vlan group (#826)

Signed-off-by: Shlomi Bitton <shlomibi@mellanox.com>

stepanblyschak commented on acl_loader/main.py:
```diff
@@ -676,7 +676,14 @@ def show_rule(self, table_name, rule_id):
 
         def pop_priority(val):
-            priority  = val.pop("PRIORITY")
+            priority = 0
```
why it is assumed to be 0 by default?

madhanmellanox replied:
This is handled in he next commit.

stepanblyschak commented on acl_loader/main.py:
```diff
@@ -676,7 +676,14 @@ def show_rule(self, table_name, rule_id):
         def pop_priority(val):
-            priority  = val.pop("PRIORITY")
+            priority = 0
+            for key in dict(val):
```
what if we will do:
```python
val = {k.upper(): v for k, v in val}
```
And fix it for any ACL rule attribute?

madhanmellanox replied:
This may not be required to set for all the ruleprops of ACL rule.

nazariig committed:
[fwutil]: Fix component table layout. (#831)

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

Commit:
Handled setting priority to None instead of 0, because 0 is not default value of ACL priority and also handled printing N/A if priority is not set in ACL rule

bsun-sudo committed:
[ntp]: support "show ntp" with mgmt vrf based on linux os version (#858)

Co-authored-by: Bing Sun <Bing_Sun@dell.com>

Junchao-Mellanox committed:
[Mellanox] Fix thermal control issue: use natural sort for fan status and thermal status (#836)

* [thermal fix] use natural sort for fan status and thermal status
* [thermal fix] set fan status to N/A when fan is removed
* Adjust header name for show platform temperature output

SuvarnaMeenakshi committed:
[multi-asic]: Update reload of systemd services to support multi-asic platforms (#856)

* Update stop, reset failed status and restart of systemd services
to support multi-asic platforms.

* Create function to avoid code duplication.

* Fixed errors due to pervious commit and review comments.

* Minor update to fix spacing.

* Minor update to fix spacing.

* Minor update to fix spacing.

* For multi asic platform updated logic of stopping/restarting of
services to ensure that the right instances are stopped and
restarted if a service is both global and multi-instance.

* Fixed log error message with incorrect number of parameterts.

baiwei0427 committed:
[pfc] Add command line to enable/disable/show PFC (#796)

Current PFC cli only allows operators to enable/disable per-port asymmetric PFC. I extend add new commands to 1) enable/disable PFC on a given priority of a given port and 2) check PFC configuration. I also link all these commands to show and config utilities.

Signed-off-by: Wei Bai baiwei0427@gmail.com
Co-authored-by: lguohan <lguohan@gmail.com>

neethajohn committed:
[watermarkstat] Fix issue of fields overwritten before display (#862)

Fix the bug where the unicast/multicast queue watermark display was incorrect

- How I did it
Calculate the header list dynamically for PG and queue display
Display the correct queue numbers in the output instead of always starting the index from 0
Calculate the display position relative to the starting index

Signed-off-by: Neetha John <nejo@microsoft.com>

jleveque committed:
Explicitly specify command names with underscores (continued) (#852)

shlomibitton committed:
Fixes bug for PFCWD feature parameters (#838)

What I did

The feature allows setting 'detection_time', 'restoration_time' and 'pollling_interval' PFCWD to an interface. The 'pollling_interval' must be lower than 'detection_time' and 'restoration_time'.
The fix is checking if there is a lower value of 'detection_time' or 'restoration_time' than the 'pollling_interval' value entered by the user in config DB, if yes exit with error code 1.

How I did it
Checking the config DB for interfaces PFCWD values.

How to verify it
Try adding 'pollling_interval' greater than one of the values of PFCWD interfaces.

Previous command output (if the output of a command-line utility has changed)
No Output.

New command output (if the output of a command-line utility has changed)
unable to use polling_interval = #ms, value is bigger or equal to the minimum in PFCWD table

jleveque committed:
Fix Python warnings (#867)

- Fix Python warnings flagged by "LGTM" plugin.
- Also make all 'except' statements Python3-compliant

jleveque committed:
Fix more Python warnings (#869)

* Fix "First parameter of a method is not named 'self'" warnings

* Fix "Module is imported with 'import' and 'import from'" warning

* Fix 'Testing equality to None' warnings

* Clean up 'Unused import' warnings

shlomibitton committed:
[showtech]: dump docker stats (#864)

* Add docker to 'show techsupport' command

Signed-off-by: Shlomi Bitton <shlomibi@mellanox.com>

jleveque committed:
Fix even more Python warnings (#873)

lguohan committed:
[doc]: add ltgm and jenkins badge (#875)

Signed-off-by: Guohan Lu <lguohan@gmail.com>

shlomibitton committed:
Add platform options to 'show techsupport' command (#865)

Signed-off-by: Shlomi Bitton <shlomibi@mellanox.com>

nazariig committed:
[fwutil]: Update Command-Reference.md. (#876)

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

Commit:
[config] Implement a process level lock (#857)

Changes:
1.) Implement a class, which uses hsetnx for lock.
2.) lock is expired within timeout period or will be released by owner.
3.) After -y prompt, lock is reacquired, because timer could have expired,
    before user enters yes.
Signed-off-by: Praveen Chaudhary pchaudhary@linkedin.com

Staphylo committed:
[generate_dump] Dump systemd specific information (#602)

- dump: dumps the entire state of all the units
 - blame: provides startup time for each service
 - plot: generate a svg of the boot order

Co-authored-by: lguohan <lguohan@gmail.com>

lguohan committed:
Revert "[config] Implement a process level lock (#857)" (#882)

This reverts commit 3f651dc0fa15d686d01ba0aa5364f7a435b6fd1c.

tsvanduyn committed:
[show][bgp] Use only 'show ip bgp' as the base and use bgp_frr_v4 file for FRR routing stack (#884)

Co-authored-by: Travis Van Duyn <trvanduy@microsoft.com>

dzhangalibaba committed:
[Vxlan] : adding show vnet/vxlan cmds (#880)

* [Vxlan] : adding show vnet/vxlan cmds

tsvanduyn committed:
[show] Add 'ip/ipv6 bgp network' commands (#888)

Co-authored-by: Travis Van Duyn <trvanduy@microsoft.com>

yxieca committed:
[fast reboot] set a fast-reboot DB flag (#887)

- This flag is currently no functional impact with just this change.
- This flag is intended to coordinate fast-reboot shutdown path.
- This flag enables moving fast-reboot per service knowledge to service scripts.
  which unblocks moving warm-reboot per service knowledge to service scripts.
- This flag is consistent with the bootup flag we setting on the boot up path.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

tahmed-dev committed:
[utility] Filter FDB entries (#890)

* [utility] Filter FDB entries

FDB table can get large due to VM creation/deletion which cause
fast reboot to slow down. This utility fitlers FDB entries based on
current MAC entries in the ARP table.

signed-off-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

rajendra-dendukuri committed:
Allow show ztp to display non-sensitive information visible to non-root user (#872)

Removed user privileges checks. These changes fix #800. The ZTP PR Azure/sonic-ztp#13 is also required for these changes to work as expected.

```
admin@sonic:~$ show ztp status
ZTP Admin Mode : True
ZTP Service : Processing
ZTP Status : IN-PROGRESS
ZTP Source : local-fs (/host/ztp/ztp_data_local.json)
Runtime : 06m 26s
Timestamp : 2020-04-08 13:08:22 UTC

ZTP Service is active

01-test-plugin: IN-PROGRESS
```

Signed-off-by: Rajendra Dendukuri <rajendra.dendukuri@broadcom.com>

rajendra-dendukuri committed:
[doc]: ZTP configuration and show commands (#866)

nazariig committed:
[fwutil]: Fix firmware update command. (#895)

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

qiluo-msft committed:
Make `config` command support abbreviation. (#893)

rkdevi27 committed:
ssd_mitigation_changes (#829)

* ssd_mitigation_changes

* ssd_mitigation_changes

* ssd_mitigation_changes

Junchao-Mellanox committed:
[psushow] Add a column to display LED color to show platform psustatus output (#886)

renukamanavalan committed:
Improved route_check tool and adopt to 20191130 image. (#898)

* Changes:
1) Add syslog support.
2) Enable forever periodic scan
3) Skip link local addresses
4) Skip eth0 routes
3) Adopt to 20191130 changes
    3.1) APPl-DB INTF_TABLE may not have IP address
    3.2) nexthop is never empty

* No logical code change. A small name change.

* 1) Adopt to 201811 -- Filter out 'lo' & 'docker0' in addition to 'eth0' as local routes
2) Ensure to read route entry w/o prefix, if not present with prefix

judyjoseph committed:
[config] Add support for multi-ASIC devices (#877)

* [Phase 1] Multi ASIC config command changes, db_mgrator.py script updates for handing namespace.

* Fixes and comment updates

* Comments addressed + added support for user to input the config files per namespace also.

* Updates per comments + based on the updated SonicV2Connector/ConfigDBConnector class design

* Review comments update.

* Help string updated for config save/reload/load

jleveque committed:
[show] Fix abbreviations for 'show ip bgp ...' commands (#901)

Commit:
addressed code review comments against Azure master

prsunny committed:
[Vnet] Fix NameError for 'swsssdk' and align output (#902)

* Fix NameError for 'swsssdk', align neig output

arlakshm committed:
[config] Support load_minigraph command for multi NPU platform (#896)

- Modify the load_minigraph command handler to support multi NPU platforms

tsvanduyn committed:
[show] Add 'errors' and 'rates' subcommands to 'show interfaces counters' group (#900)

Co-authored-by: Travis Van Duyn <trvanduy@microsoft.com>

jleveque committed:
[config] Add 'interface transceiver' subgroup with 'lpmode' and 'reset' subcommands (#904)

jleveque committed:
Remove dependency on click-default-group package (#903)

jleveque committed:
[config] Log invocation of config commands to syslog (#259)

yangshp1987 committed:
[show] Add `ntpstat` output to `show ntp` (#861)

Signed-off-by: yangshiping <yangshiping@jd.com>

Commit:
added the fix for aclshow script and add a unit test case for the fix

Commit:
Merge branch 'aclpriority' of https://github.com/madhanmellanox/sonic-utilities into aclpriority

madhanmellanox commented:
This is an internal PR and Azure PR corresponding to this PR got merged, so closing this one.
